test/e2e/k8s_util.go

```diff
@@ -941,7 +942,17 @@ func (k *KubernetesUtils) Cleanup(namespaces []string) {
 
 	for _, ns := range namespaces {
 		log.Infof("Deleting test Namespace %s", ns)
-		if err := k.clientset.CoreV1().Namespaces().Delete(context.TODO(), ns, metav1.DeleteOptions{}); err != nil {
+		// Using Poll to ensure namespace is deleted successfully, because sometimes
+		// it takes more time to delete a namespace and if that action is overridden by
+		// some new task then the contents of the namespace are not deleted properly and,
+		// a race condition may arise for further tests.
+		err := wait.Poll(5*time.Second, 10*time.Second, func() (done bool, err error) {
+			if err := k.clientset.CoreV1().Namespaces().Delete(context.TODO(), ns, metav1.DeleteOptions{}); err != nil {
```

Comment:
There is some misunderstanding. "Poll for the Namespace deletion" means it should poll the existence of the namespace after deleting it, like https://github.com/antrea-io/antrea/blob/0e7e0ba6d6c56b78ee8ac0b4bf02946b3d40a721/test/e2e/framework.go#L625-L645. I think you could just use that function.